src/query/generated/proto/rpcpb/query.proto

```diff
@@ -90,3 +92,44 @@ message M3Segment {
 	int64 startTime = 3;
 	int64 blockSize = 4;
 }
+
+message SearchRequest {
+	oneof matchers {
+		TagMatchers tagMatchers     = 1;
+	}
+	SearchType type               = 2;
+	repeated bytes filterNameTags = 3;
```

Comment:
nit: Maybe we could create another message called `SearchRequestOptions` that contains this `filterNameTags` field?  That way in the future we can combine all the search options fields together in one logical group.

Not a big deal, but from an API design angle it's a little more clean.